### PR TITLE
add vendor-interfaces model/table

### LINKED ISSUE #333: Initialize vendor_interfaces model for vendor load
Using https://github.com/sul-dlss/libsys-airflow/pull/329/files as an example (or maybe PR #352), initialize the "vendor_interfaces" model based on https://docs.google.com/document/d/1xjbxiFWHT3THgaYC6mP1MS6VKoQKA2GWKey1TE0Y5cE/edit#heading=h.6qi50lyx2uxz

See also https://github.com/sul-dlss/libsys-airflow/wiki/Vendor-Load:--Database-Model

This model is required to capture the vendor_interface configuration options described in the design document above per vendor_interface.

- [x] Add vendor_interfaces model under the data management plugin

### PR DESCRIPTION
**Pull request by ndushay**
Closes #333.

My main questions:

- are tests needed?
- does anything need to happen to docker postgres container so it has the latest migration applied?
- @aaron-collier the ticket talks about code to use the model and populate the uuid and so on.  Could that be a separate ticket?

I meant to include this to show it is working:

```
You are now connected to database "vendor_loads" as user "airflow".
vendor_loads=# \d+ vendor_interfaces
                                                                            Table "public.vendor_interfaces"
              Column               |          Type          | Collation | Nullable |                    Default                    | Storage  | Compression | Stats target | Description 
-----------------------------------+------------------------+-----------+----------+-----------------------------------------------+----------+-------------+--------------+-------------
 id                                | integer                |           | not null | nextval('vendor_interfaces_id_seq'::regclass) | plain    |             |              | 
 vendor_id                         | integer                |           |          |                                               | plain    |             |              | 
 display_name                      | character varying(50)  |           | not null |                                               | extended |             |              | 
 folio_interface_uuid              | character varying(36)  |           | not null |                                               | extended |             |              | 
 folio_data_import_profile_uuid    | character varying(36)  |           | not null |                                               | extended |             |              | 
 folio_data_import_processing_name | character varying(50)  |           | not null |                                               | extended |             |              | 
 file_pattern                      | character varying(250) |           |          |                                               | extended |             |              | 
 remote_path                       | character varying(250) |           |          |                                               | extended |             |              | 
 processing_dag                    | character varying(50)  |           |          |                                               | extended |             |              | 
 processing_delay_in_days          | integer                |           |          |                                               | plain    |             |              | 
 active                            | boolean                |           | not null |                                               | plain    |             |              | 
Indexes:
    "vendor_interfaces_pkey" PRIMARY KEY, btree (id)
    "vendor_interfaces_folio_data_import_profile_uuid_key" UNIQUE CONSTRAINT, btree (folio_data_import_profile_uuid)
    "vendor_interfaces_folio_interface_uuid_key" UNIQUE CONSTRAINT, btree (folio_interface_uuid)
Foreign-key constraints:
    "vendor_interfaces_vendor_id_fkey" FOREIGN KEY (vendor_id) REFERENCES vendors(id)
Access method: heap

vendor_loads=# \d+ vendors
                                                                            Table "public.vendors"
            Column            |            Type             | Collation | Nullable |               Default               | Storage  | Compression | Stats target | Description 
------------------------------+-----------------------------+-----------+----------+-------------------------------------+----------+-------------+--------------+-------------
 id                           | integer                     |           | not null | nextval('vendors_id_seq'::regclass) | plain    |             |              | 
 display_name                 | character varying(50)       |           | not null |                                     | extended |             |              | 
 folio_organization_uuid      | character varying(36)       |           | not null |                                     | extended |             |              | 
 vendor_code_from_folio       | character varying(36)       |           | not null |                                     | extended |             |              | 
 acquisitions_unit_from_folio | character varying(36)       |           | not null |                                     | extended |             |              | 
 has_active_vendor_interfaces | boolean                     |           | not null |                                     | plain    |             |              | 
 last_folio_update            | timestamp without time zone |           | not null |                                     | plain    |             |              | 
Indexes:
    "vendors_pkey" PRIMARY KEY, btree (id)
    "vendors_display_name_key" UNIQUE CONSTRAINT, btree (display_name)
    "vendors_folio_organization_uuid_key" UNIQUE CONSTRAINT, btree (folio_organization_uuid)
    "vendors_vendor_code_from_folio_key" UNIQUE CONSTRAINT, btree (vendor_code_from_folio)
Referenced by:
    TABLE "vendor_interfaces" CONSTRAINT "vendor_interfaces_vendor_id_fkey" FOREIGN KEY (vendor_id) REFERENCES vendors(id)
Access method: heap
```

### CODE REVIEW

File: libsys_airflow/plugins/vendor/models.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
 from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Text, Boolean
 from sqlalchemy.orm import declarative_base, relationship, Session
-from airflow.providers.postgres.hooks.postgres import PostgresHook
+#from airflow.providers.postgres.hooks.postgres import PostgresHook
```
Comment (jmartin-sul) [inline libsys_airflow/plugins/vendor/models.py:3]: cruft?  also i think this is one of the linter's complaints.

**Comment by aaron-collier** [comment]
@ndushay 

- I don't necessarily think we need basic model tests until (or if) we add any specific functionality
- No - the migrations will have to be run manually (much like rails) but we'll ticket adding the migrations to capistrano for development.
- Yes, good catch. I think that should be a separate ticket (or a few) - this was mostly a copy paste brain dump so I think you're good here.

**Review by jmartin-sul — COMMENTED** [review]
i need to glance at the alembic docs and the other alembic PR to leave a real review, but to the (first two) questions:

>  * are tests needed?

thinking about our usual rails practices, i don't know that we usually add tests for migrations exactly, but sometimes we do add some basic tests to make sure the model behaves as expected?  since this adds a new model class, perhaps some basic tests around things like uniqueness and null-ability might be warranted?

> * does anything need to happen to docker postgres container so it has the latest migration applied?

my recollection from the monday clumping session on alembic and DB migrations (and some post-standup discussion yesterday):
* locally, the user is expected to invoke `poetry run alembic upgrade head` to apply any migrations that have yet to be applied locally, a la `rails db:migrate` usage in our rails dev setups.  the [docker-compose.yaml exposes the postgres ports](https://github.com/sul-dlss/libsys-airflow/blob/main/docker-compose.yaml#L109-L110) to the host, so should be runnable from the laptop command line w/o needing to drop into the container?  looks like the initial alembic PR [added that to the README](https://github.com/sul-dlss/libsys-airflow/pull/352/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R134).
* either @aaron-collier or @jermnelson was going to write a ticket about adding to the capistrano hooks, such that migrations would be auto-run against the (ops managed/not containerized) postgres instances we use in our deployed envs.  but i don't see such a ticket yet.

**Comment by ndushay** [comment]
I fixed the lint errors;  I also put some output from psql showing the new table.  Ready for re-review.

(I leave it to Aaron to make the capistrano ticket and such.)

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**